### PR TITLE
Add DCUI and automated disk/network/password setup to OVA (#3913)

### PR DESCRIPTION
**Pull request by frapposelli**

### CODE REVIEW

File: installer/ovatools/ovfenv/main.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ func main() {
 			Value: "",
 			Usage: "Get single OVF property",
 		},
+		cli.BoolFlag{
+			Name:  "dump",
+			Usage: "Dump the OVF Environment XML",
```
Comment (vburenin) [inline installer/ovatools/ovfenv/main.go]: dump and exit?

Comment (frapposelli) [inline installer/ovatools/ovfenv/main.go]: yep, dumps the XML and quits, useful for debugging when the go parser fails to handle proper xml 😄

File: installer/ovatools/vic-ova-ui/main.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+)
+
+const (
+	VT_ACTIVATE   = 0x5606
+	VT_WAITACTIVE = 0x5607
+)
+
+func main() {
+
+	if runtime.GOOS == "linux" {
```
Comment (dougm) [inline installer/ovatools/vic-ova-ui/main.go]: There's other linux specific stuff in here, maybe just skip the check?  Or if we want it to run on more than linux, could move the platform specifics to main_linux.go, main_darwin.go, etc.

File: installer/ovatools/vic-ova-ui/main.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	}
+	defer ui.Close()
+
+	gray := ui.ColorRGB(1, 1, 1)
+	blue := ui.ColorCyan
+
+	// Check if we're running inside a VM
+	isVM, err := vmcheck.IsVirtualWorld()
+	if err != nil {
+		fmt.Printf("error: %s\n", err.Error())
```
Comment (dougm) [inline installer/ovatools/vic-ova-ui/main.go]: errors to os.Stderr instead of Stdout?

File: installer/ovatools/vic-ova-ui/main.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+		Value string `xml:"value,attr"`
+	}
+
+	type propertySection struct {
+		Property []property `xml:"Property"`
+	}
+
+	var environment struct {
+		XMLName         xml.Name        `xml:"Environment"`
+		PropertySection propertySection `xml:"PropertySection"`
```
Comment (dougm) [inline installer/ovatools/vic-ova-ui/main.go]: There is a an ovf/env package in govmomi, maybe we could use/extend that? https://github.com/vmware/govmomi/blob/master/ovf/env.go#L51

File: installer/ovatools/vic-ova-ui/main.go (hunk 2 above)
Comment (frapposelli) [inline installer/ovatools/vic-ova-ui/main.go]: I thought about that, reality is that the only use case for this is to run on a Linux os, the check is just to help when doing interactive tests on a non-linux platform.

File: installer/ovatools/vic-ova-ui/main.go (hunk 3 above)
Comment (frapposelli) [inline installer/ovatools/vic-ova-ui/main.go]: Good call 😄

File: installer/ovatools/vic-ova-ui/main.go (hunk 4 above)
Comment (frapposelli) [inline installer/ovatools/vic-ova-ui/main.go]: Interesting, I'll definitely add a story to extend that and use it here.

File: installer/ovatools/vic-ova-ui/main.go (hunk 2 above)
Comment (frapposelli) [inline installer/ovatools/vic-ova-ui/main.go]: Actually, I'll make it linux, it's going to be easier to maintain going forward anyway

**Pull request merged**